.buildkite/pipeline.tests-production-canary.yaml

```diff
@@ -0,0 +1,23 @@
+# These pipeline steps constitute the quality gate for fleet server within the production canary environment.
+# A failure in this pipeline build will prevent further progression to the subsequent stage.
+
+steps:
+  - label: ":cookie: 1h bake period before continuing promotion"
+    command: "sleep 3600"
+  - wait
+  - label: ":rocket: Run observability gates"
+    trigger: "serverless-quality-gates"
```

Comment:
unfortunately these quality gates don't work in production, I tried to enable before: https://github.com/elastic/serverless-quality-gates/blob/main/observability/execute-query.sh#L27

I think we have to come up with another check as quality gate, maybe we can trigger the e2e tests like kibana: https://github.com/elastic/kibana/blob/main/.buildkite/pipelines/quality-gates/pipeline.tests-production-canary.yaml#L7

Reply:
Yes, we get an alert if the SLO breaks during the baking period in canary.
AFAIK we can manually stop the buildkite job that runs the wait, so the promotion doesn't continue. What I'm not sure about if production-noncanary gets automatically locked or we have to manually lock it, I can ask around to find out.